gnomad_qc/v4/create_release/calculate_variant_statistics.py

```diff
@@ -24,188 +51,218 @@
+    # Add extra Allele Count and Allele Type annotations to variant MatrixTable,
+    # according to Hail standards, to help their computation.
+    mt = mt.annotate_rows(
+        **vmt_sample_qc_variant_annotations(global_gt=mt.GT, alleles=mt.alleles)
     )
 
-    # Add extra Allele Count and Allele Type annotations to variant Matrix
-    # Table, according to Hail standards, to help their computation.
-    ac_and_atype = vmt_sample_qc_variant_annotations(
-        global_gt=vmt.GT, alleles=vmt.alleles
-    )
-    vmt = vmt.annotate_rows(**ac_and_atype)
-
+    keep_annotations = ["freq", "filters", "region_flags"]
     if by_csqs:
-        ht = filter_vep_transcript_csqs(
-            ht,
+        annotation_ht = filter_vep_transcript_csqs(
+            annotation_ht,
             synonymous=False,
             canonical=vep_canonical,
             mane_select=vep_mane,
         )
+        annotation_ht = get_most_severe_consequence_for_summary(annotation_ht)
+        keep_annotations.extend(["most_severe_csq", "lof"])
 
-        ht = get_most_severe_consequence_for_summary(ht)
-
-        ht = ht.select("freq", "filters", "region_flags", "most_severe_csq", "lof")
-
-    # NOTE: not size efficient, but we are never checkpointing THIS I don't believe
-    vmt = vmt.annotate_rows(**ht[vmt.row_key])
+    # Annotate the MT with the needed annotations.
+    annotation_ht = annotation_ht.select(*keep_annotations).checkpoint(
+        hl.utils.new_temp_file("annotation_ht", "ht")
+    )
+    mt = mt.annotate_rows(**annotation_ht[mt.row_key])
 
-    arg_dict = {"all_variants": ~hl.is_missing(vmt.locus)}
+    filter_expr = {"all_variants": True}
 
     if pass_filters:
-        arg_dict.update({"pass_filters": hl.len(vmt.filters) == 0})
-    if ukb_regions:
-        arg_dict.update({"ukb_regions": ~vmt.region_flags.outside_ukb_capture_region})
-    if broad_regions:
-        arg_dict.update(
-            {"broad_regions": ~vmt.region_flags.outside_broad_capture_region}
+        filter_expr["pass_filters"] = hl.len(mt.filters) == 0
+    if ukb_capture:
+        filter_expr["ukb_capture"] = ~mt.region_flags.outside_ukb_capture_region
+    if broad_capture:
+        filter_expr["broad_capture"] = ~mt.region_flags.outside_broad_capture_region
+    if ukb_capture and broad_capture:
+        filter_expr["ukb_broad_capture_intersect"] = (
+            filter_expr["ukb_capture"] & filter_expr["broad_capture"]
         )
-    if ukb_regions and broad_regions:
-        arg_dict.update(
-            {
-                "ukb_and_broad_regions": (
-                    ~vmt.region_flags.outside_ukb_capture_region
-                ) & (~vmt.region_flags.outside_broad_capture_region)
-            }
+        filter_expr["ukb_broad_capture_union"] = (
+            filter_expr["ukb_capture"] | filter_expr["broad_capture"]
         )
     if rare_variants:
-        arg_dict.update({"rare_variants": vmt.freq[0].AF < 0.001})
+        # TODO: Maybe make this a parameter?
```

Comment:
@matren395 @KoalaQin thoughts on this TODO?

Reply:
could very easily do, but Sam's request was <0.1% (https://app.zenhub.com/workspaces/gnomad-5f4d127ea61afc001d6be50b/issues/gh/broadinstitute/gnomad_production/1154) so I'm cool leaving it as well.